engine/src/core/graphics/rendering/rendering_device/batch_2d.cpp

```diff
@@ -0,0 +1,158 @@
+//
+// Created by blinku on 23.11.2019.
+//
+
+#include "batch_2d.h"
+#include "graphics/rendering/camera_2d.h"
+#include "graphics/rendering/sprite.h"
+#include <cstring>
+
+namespace Bald::Graphics {
+    static constexpr uint32_t MAX_SPRITES = 10000;
+    static constexpr uint32_t SPRITE_VERTEX_DATA_SIZE = sizeof(SpriteVertexData);
+    static constexpr uint32_t SPRITE_SIZE = 4 * SPRITE_VERTEX_DATA_SIZE;
+    static constexpr uint32_t INDICES_SIZE = 6;
+    static constexpr uint32_t MAX_VERTICES = SPRITE_SIZE * MAX_SPRITES;
+    static constexpr uint32_t MAX_INDICES = INDICES_SIZE * MAX_SPRITES;
+
+    Batch2D::Batch2D() :
+        m_MaxVertices(MAX_VERTICES),
+        m_UsedVertices(0),
+        m_MaxIndices(MAX_INDICES),
+        m_UsedIndices(0),
+        m_QuadVAO(nullptr),
+        m_QuadVBO(nullptr) {
+        VertexBufferLayout layout = {
```

Comment:
I may be wrong here but I thought buffers were created like this to let the user shape his data in his way. If we do it like this it is hardcoded.

Reply:
Yes, however, our renderer currently supports only sprites/quads. If you want to do OpenGL yourself, you can do it with these abstractions.